src/fiatconnect/saga.ts

```diff
@@ -445,14 +459,54 @@ export function* _getSpecificQuote({
     providerIds: [providerId],
   })
   const normalizedQuotes = normalizeFiatConnectQuotes(flow, quotes)
-  const normalizedQuote = normalizedQuotes.find((q) => q.getFiatAccountType() === fiatAccountType)
+
+  // If no account was provided, we need to check cached accounts for a possible match, and fetch the obfuscated account
+  // from the selected provider.
+  let selectedFiatAccount = fiatAccount
+  if (!selectedFiatAccount) {
+    const cachedFiatAccounts: CachedFiatAccountUse[] = yield select(cachedFiatAccountUsesSelector)
+    // If defined, cachedFiatAccount is a Fiat Account whose providerId and fiatAccountType match one of our quotes.
+    const cachedFiatAccount = cachedFiatAccounts.find((a) =>
+      normalizedQuotes.find(
+        (q) => q.getFiatAccountType() === a.fiatAccountType && providerId === a.providerId
```

Comment:
what if q and a have the same fiat account type and different fiat account schemas? seems like a pretty important case to consider

Reply:
D'oh -- that's a great point; thanks for the catch!

Reply:
We actually don't save the fiat account schema used in `cachedFiatAccountUses`, which makes performing this check somewhat impossible at the moment..